### PR TITLE
Reborn

### PR DESCRIPTION
**Pull request by michaelkuty**

### CODE REVIEW

File: elephantagenda/api.py [hunk 1]
```diff
@@ -17,5 +18,7 @@ def events(request):
         events = Event.objects.all()
     else:
         events = Event.objects.upcoming()
-        
-    return HttpResponse(serializers.serialize('json', events, ensure_ascii=False), mimetype="text/javascript")
\ No newline at end of file
+
+    return HttpResponse(
+        serializers.serialize('json', events, ensure_ascii=False),
+        mimetype="text/javascript")
```
Comment (matthiask) [inline elephantagenda/api.py]: `content_type`, not `mimetype`

File: AUTHORS [hunk 2]
```diff
@@ -0,0 +1 @@
+Simon Bächler <sb@feinheit.ch>
```
Comment (matthiask) [inline AUTHORS:2]: Feel free to add yourself as well!

File: elephantagenda/apps.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+from django.apps import AppConfig
+
+
+class AgendaConfig(AppConfig):
+    name = 'elephantagenda'
+    verbose_name = "Agenda"
```
Comment (matthiask) [inline elephantagenda/apps.py]: Please make `verbose_name` translatable (using `ugettext_lazy`) and keep it lowercase.

File: elephantagenda/backends/agenda/models.py [hunk 4]
```diff
@@ -136,68 +138,77 @@ class Meta:
...
-        
+            raise ValidationError(
+                _('The Event cannot end before start (Start date <= End date)'))
+
     @property
     def location(self):
         return self.venue.name
-    
+
     @models.permalink
```
Comment (matthiask) [inline elephantagenda/backends/agenda/models.py:159]: Django recommends the use of `reverse` instead of `permalink`

File: elephantagenda/urls.py [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-        }, name='agenda_event_detail'),
-    url(r'^archive/$', 'django.views.generic.list_detail.object_list', {
-        'queryset': Event.objects.past(),
-        'paginate_by': 20,
-        }, name='agenda_event_list'),
-                       
-    url('^api/events/', events, name='agenda_api_events'),
-)
\ No newline at end of file
+                       url(r'^$', 'django.views.generic.list_detail.object_list', {
```
Comment (matthiask) [inline elephantagenda/urls.py]: Please switch to generic CBV (`ListView` etc.)

**Comment by michaelkuty** [comment]
@matthiask What you think about making event as feincms content type with regions and remove description, picture etc just a vanilla model with extra regions in default module ?

File: elephantagenda/views.py [hunk 6]
```diff
@@ -14,10 +13,15 @@ def event_list(request, filter=None):
         events = Event.objects.past().filter(language=short_language_code)
     else:
         events = Event.objects.active().filter(language=short_language_code)
-    
-    return render_to_string('agenda/event_list.html', {'object_list' : events}, RequestContext(request))
+
+    return render_to_string(
```
Comment (matthiask) [inline elephantagenda/views.py]: Use `render` instead, so that there's no need to use `RequestContext` directly.

File: setup.cfg [hunk 7]
```diff
@@ -0,0 +1,30 @@
+[metadata]
+name = feincms-elephantagenda
+summary = A versatile agenda module for FeinCMS.
+description-file =
+    README.rst
+author = Simon Bächler
+author-email = mail@michaelkuty.eu
```
Comment (matthiask) [inline setup.cfg]: Did you leave your mail address in here intentionally?

File: setup.cfg [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+version = 1.0.0
+home-page = https://github.com/feinheit/feincms-elephantagenda.git
+classifier =
+    Development Status :: 5 - Production/Stable
+    Framework :: Django
+    Intended Audience :: Developers
+    License :: OSI Approved :: BSD License
+    Operating System :: OS Independent
+    Programming Language :: Python
+    Programming Language :: Python :: 2.6
```
Comment (matthiask) [inline setup.cfg]: From my point of view – drop 2.6 already, but I don't care either way.

**Comment by matthiask** [comment]
Wow. Thank you! Added a few comments.

File: setup.cfg (hunk 7 above)
Comment (michaelkuty) [inline setup.cfg]: I was lazy to search Simon mail

Comment (matthiask) [inline setup.cfg]: That's fine. Either leave your mail in there, or add `dev@feinheit.ch`

**Comment by michaelkuty** [comment]
Ok, @matthiask this is usable base for next work, I still thinking about content model with regions as new backend, but this need some clients which another client who wanted extra content.

TODO: facebook, gapis but I let it for another contributors now

File: elephantagenda/api.py [hunk 9]
```diff
@@ -17,5 +18,7 @@ def events(request):
         events = Event.objects.all()
     else:
         events = Event.objects.upcoming()
-        
-    return HttpResponse(serializers.serialize('json', events, ensure_ascii=False), mimetype="text/javascript")
\ No newline at end of file
+
+    return JsonResponse(
```
Comment (matthiask) [inline elephantagenda/api.py:21]: It seems you're serializing an already serialized representation here?

File: elephantagenda/templates/agenda/event_list.html [hunk 10]
```diff
@@ -6,6 +6,7 @@
             <th>{% trans "Location" %}</th>
             <th>{% trans "Start" %}</th>
             <th>{% trans "End" %}</th>
+            <th>{% trans "Short Desription" %}</th>
```
Comment (matthiask) [inline elephantagenda/templates/agenda/event_list.html]: Typo (Des c ription)

**Comment by matthiask** [comment]
Thanks again! Awesome work. Now we're almost entering the stage of nitpicking — will merge soon. I think the `JsonResponse` change is incorrect, apart from that we're good to go.

File: elephantagenda/api.py (hunk 9 above)
Comment (michaelkuty) [inline elephantagenda/api.py:21]: You are absolutely right !

**Comment by michaelkuty** [comment]
NO this is the feedback which I really need, fast and concrete !

**Pull request merged**